### ISSUE TITLE
Using multiple endpoints 

### ISSUE THREAD

**Issue by gauravcmu** [gambol99/go-marathon#230]
Hi, I am running into one issue with creating marathon client using multiple endpoints. My goal is to use two endpoints, in HA mode, allowing failure handling. The problem I observe is that if the first marathon endpoint in the URL is not available due to some reason, my client never falls back to second provided in the client URL. This might be mostly a usage issue, so want to understand whats missing.  

I wrote a simple program to show how I am using it. Providing config URL "http://192.168.10.101:8080,http://192.168.10.100:8080", where marathon is running on "http://192.168.10.100:8080", i.e. the second endpoint. I observe that in the first client creation, followed by Applications() call, I see error with getsockopt() as it tries to go to the first endpoint only. It seems it never tried the second in the URL string. 

After this I try creating the client with URL "http://192.168.10.100:8080" and don't see any issue. 

```
package main

import (
	"fmt"
	"github.com/gambol99/go-marathon"
)

func main() {

	//Try creating connection to marathon hosts - first URL won't work.
	//Testing if NewClient() creates falls back to the second URL, i.e. http://192.168.10.100:8080
	testConfig := marathon.NewDefaultConfig()
	testConfig.URL = "http://192.168.10.101:8080,http://192.168.10.100:8080"

	fmt.Printf("\nTry creating connection with two endpoint in URL config...\n")
	mclient, err := marathon.NewClient(testConfig)
	if err != nil {
		fmt.Printf("marathon client creation failed: %v\n", err)
		_ = mclient
	}
	apps, err := mclient.Applications(nil)
	if err != nil {
		fmt.Printf("marathon getting apps failed: %v\n", err)
	} else {
		fmt.Printf("apps %+v", apps)
	}

	fmt.Printf("\nTry creating connection with URL in config...\n")
	//Try creating client with one working URL http://192.168.10.100:8080
	testConfig2 := marathon.NewDefaultConfig()
	testConfig2.URL = "http://192.168.10.100:8080"
	mclient2, err := marathon.NewClient(testConfig2)
	if err != nil {
		fmt.Printf("marathon client2 creation failed: %v\n", err)
		_ = mclient2
	}

	apps, err = mclient2.Applications(nil)
	if err != nil {
		fmt.Printf("marathon getting apps failed: %v\n", err)
	} else {
		fmt.Printf("apps %+v", apps)
	}

}

```
Here is the output showing what problem I observe:- 
```

$./marathonclient

Try creating connection with two endpoint in URL config...
marathon getting apps failed: Get http://192.168.10.101:8080/v2/apps: dial tcp 192.168.10.101:8080: getsockopt: no route to host

Try creating connection with URL in config...
apps &{Apps:[{ID:test Cmd:<nil> Args:<nil> Constraints:0xc8200aad80 Container:0xc8200aaf40 CPUs:0.5 Disk:0xc82000b7e0 Env:0xc820024058 Executor:0xc82000b7f0 HealthChecks:0xc8200ab120 Instances:0xc82000b7c0 Mem:0xc82000b7d0 Tasks:[] Ports:[30783 10000] PortDefinitions:<nil> RequirePorts:0xc82000b87a BackoffSeconds:0xc82000b880 BackoffFactor:0xc82000b890 MaxLaunchDelaySeconds:0xc82000b8a0 Deployments:[] Dependencies:[] TasksRunning:1 TasksStaged:0 TasksHealthy:0 TasksUnhealthy:0 User: UpgradeStrategy:0xc82000bad0 Uris:0xc8200aadc0 Version:2016-11-03T09:42:17.266Z VersionInfo:0xc8200ab280 Labels:0xc820024060 AcceptedResourceRoles:[] LastTaskFailure:<nil> Fetch:0xc8200aae40 IPAddressPerTask:<nil>}]}
```

So, my question is, to make the multiple endpoints work, meaning allowing fail over to the other one in the URL string, what changes do I need to make. I am probably missing something here. 

Thanks.

**Comment by timoreimann** [comment]
I don't think you are doing anything wrong. There's actually a bug in the logic responsible for failing over to other nodes that we have discovered fairly recently. It's being tracked in #212.

I'll try to get some more focus on the issue.

Thanks for reporting!

**Comment by gauravcmu** [comment]
@timoreimann - thank you, appreciate your quick response.

**Comment by timoreimann** [comment]
@gauravcmu this should be fixed by #231 now. If you are still experiencing issues, please let us know.
